### PR TITLE
Test/bypass on docs only change

### PR DESCRIPTION
**Pull request by jboreiko**
Using golden time to help out those docs contributors (also keeping docs PRs out of my way in the queue when not necessary)

Special shout out to our docs master and all mighty product manager @rhero

I have tested that this can detect when non doc changes have occurred.   I haven't been able to test that it behaves properly with docs only changes from origin/develop because as you can imagine this code is always part of the change and therefore requires the tests...

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/1200)

### CODE REVIEW

**Comment by rhero** [comment]
Woot! If there's no way to guarantee it doesn't kick-off builds for docs only, are we just going to merge it and see what happens?

Also, is it having issues with circle?

**Comment by jboreiko** [comment]
For got I also submitted a rename with this pr, fixing it now

File: scripts/circle-ci/check-only-docs-changes.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+# Returns true if only the docs have been modified
+
+set -x
+
+DOCS_REPO="docs/"
+
+FILES_CHANGED=$(git log --name-only --pretty=format: origin/develop..HEAD)
```
Comment (SerialVelocity) [inline scripts/circle-ci/check-only-docs-changes.sh]: If something merges into develop, this query will no longer work and will cause full builds

File: scripts/circle-ci/check-only-docs-changes.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+for word in $FILES_CHANGED
```
Comment (SerialVelocity) [inline scripts/circle-ci/check-only-docs-changes.sh]: `for file`?

File: scripts/circle-ci/run-circle-tests.sh [hunk 3]
```diff
@@ -25,12 +27,22 @@ do
     CONTAINER_0_EXCLUDE_ARGS="$CONTAINER_0_EXCLUDE_ARGS -x $task"
 done
 
+# Short circuit the build if it's docs only
+./scripts/circle-ci/check-only-docs-changes.sh
```
Comment (SerialVelocity) [inline scripts/circle-ci/run-circle-tests.sh]: You can inline this into the if. Just don't use any brackets.

File: scripts/circle-ci/run-circle-tests.sh [hunk 4]
```diff
@@ -25,12 +27,22 @@ do
...
+if [ $? -eq 0 ]; then
+    if [ $CIRCLE_NODE_INDEX -eq 0 ]; then
+        checkDocsBuild
+        exit
```
Comment (SerialVelocity) [inline scripts/circle-ci/run-circle-tests.sh]: `exit $?` makes it a bit clearer that you're trying to return the exit code.

**Comment by SerialVelocity** [comment]
Do we want something to bypass release notes if only docs have changed?

File: scripts/circle-ci/check-only-docs-changes.sh (hunk 2 above)
Comment (jboreiko) [inline scripts/circle-ci/check-only-docs-changes.sh]: lol, yeah

File: scripts/circle-ci/check-only-docs-changes.sh (hunk 1 above)
Comment (jboreiko) [inline scripts/circle-ci/check-only-docs-changes.sh]: This should will still work regardless of changes in develop, it's all references from HEAD back to devleop.  Changed it anyways to a diff using ...

File: scripts/circle-ci/run-circle-tests.sh (hunk 3 above)
Comment (jboreiko) [inline scripts/circle-ci/run-circle-tests.sh]: I thought this was more readable but can change, let me know

File: scripts/circle-ci/run-circle-tests.sh (hunk 4 above)
Comment (jboreiko) [inline scripts/circle-ci/run-circle-tests.sh]: sure

File: scripts/circle-ci/run-circle-tests.sh (hunk 3 above)
Comment (jboreiko) [inline scripts/circle-ci/run-circle-tests.sh]: changed

**Pull request merged**